test/extended/psap/psap.go

```diff
@@ -0,0 +1,155 @@
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		e2e.Logf("[== Assert if the tuned openshift-stalld created successfully ==]")
+		tunedStdOut, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("tuned", "-n", ntoNamespace).Output()
+		e2e.Logf("current tuned status is:\n%s, ", tunedStdOut)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(tunedStdOut).NotTo(o.BeEmpty())
+		o.Expect(tunedStdOut).To(o.ContainSubstring("openshift-stalld"))
+
+		//Assert if profile applied to label node with re-try
+		o.Eventually(func() bool {
+
+			appliedStatus, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("-n", ntoNamespace, "profile", firstCoreOSWorkerNodes, `-ojsonpath='{.status.conditions[?(@.type=="Applied")].status}'`).Output()
+			if err != nil || strings.Contains(appliedStatus, "False") || strings.Contains(appliedStatus, "Unknown") {
+				e2e.Logf("Failed to applied custom profile to nodes, the status is %s and %v, check again", appliedStatus, err)
+			}
+			return strings.Contains(appliedStatus, "True")
+
+		}, 5*time.Second, time.Second).Should(o.BeTrue())
+
+		e2e.Logf("[== Assert if the custom profile openshift-stalld applied to label node ==]")
+		profileStdOut, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("-n", ntoNamespace, "profile", firstCoreOSWorkerNodes, "-ojsonpath={.status.tunedProfile}").Output()
+		e2e.Logf("current profile status is [ %s ] on [ %s ]", profileStdOut, firstCoreOSWorkerNodes)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(profileStdOut).NotTo(o.BeEmpty())
+		o.Expect(profileStdOut).To(o.ContainSubstring("openshift-stalld"))
+
+		e2e.Logf("[== Check if stalld service is running ... ==]")
+		stalldStatus, err := exutil.DebugNodeRetryWithOptionsAndChroot(oc, firstCoreOSWorkerNodes, ntoNamespace, "systemctl", "status", "stalld")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(stalldStatus).To(o.ContainSubstring("active (running)"))
+
+		//Save stalld startup log and compare the debug log after 10 minutes
+		//Extract sub-string of stalld PID from stalld log as [1] [1] [239426] [239426] [239426] [239426] [239426] [239426] [239426] [239426]
+		//Compare sub-string after 10 minutes, if no new pid or extra string found, it means stalld log no change, the stall didn't restart
+		exp := regexp.MustCompile(`\[.*?\]`)
+
+		stalldLogStdOut, err := exutil.DebugNodeRetryWithOptionsAndChroot(oc, firstCoreOSWorkerNodes, ntoNamespace, "journalctl", "-u", "stalld", "--since=@"+currentDatetime)
+		e2e.Logf("stalld log is:\n%s", stalldLogStdOut)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		// Matches would contain a slice of all successive matching text
+		// based on the regular expression.
+		allMatches := exp.FindAllString(stalldLogStdOut, -1)
+		previousMatches := strings.Join(allMatches, " ")
+		e2e.Logf("All matches in previous snapshot of stalld log < %v > ", previousMatches)
+
+		//Wait for 10 minutes and check stalld log in the meantime and exit if finding stalld restarted
```

Comment:
s/if finding stalld restarted/if stalld restarted/
or 
s/if finding stalld restarted/if we found stalld restarted/

Reply:
Fixed in new code